tools/check-templates

```diff
@@ -43,11 +44,19 @@ def check_html_templates(templates, modified_only):
     # The casperjs files use HTML5 (whereas Zulip prefers XHTML), and
     # there are also cases where Casper deliberately uses invalid HTML,
     # so we exclude them from our linter.
+    templates_casperjs_filtered = filter(
+        lambda fn: ('casperjs' not in fn),
+        templates) # type: (Iterable[str])
+
     templates = filter(
-        lambda fn: ('base.html' not in fn) and ('casperjs' not in fn),
-        templates)
+        lambda fn: ('base.html' not in fn),
+        templates_casperjs_filtered)
+
```

Comment:
I feel like we can simplify this section by just having a "templates" list that we use for both checking ids and checking other stuff.  It seems like the only difference between the two is base.html.  If we want two lists, then we should rename "templates" to be more specific.   But, really, I'd just try to keep it simple here.